scalapact-core/src/main/scala/com/itv/scalapactcore/common/matching/WildCardRuleMatching.scala

```diff
@@ -109,25 +115,36 @@ object WildCardRuleMatching {
 
   private def checkFieldInAllArrayElements(next: NextArrayToMatch): List[ArrayMatchingStatus] =
     next.received.map { ra =>
-      //TODO: Missing regex!!
       MatchingRule.unapply(next.ruleAndContext.rule).map {
-        case (None, None, Some(arrayMin)) =>
-          if (ra.isArray && ra.arrayOrEmpty.length >= arrayMin) RuleMatchSuccess
+        case (None, None, Some(arrayMin)) if ra.isArray =>
+          if (ra.arrayOrEmpty.length >= arrayMin) RuleMatchSuccess
           else RuleMatchFailure
 
-        case (Some(matchType), None, Some(arrayMin)) if matchType == "type" =>
-          if (ra.isArray && ra.arrayOrEmpty.length >= arrayMin) RuleMatchSuccess
+        case (Some(matchType), None, Some(arrayMin)) if ra.isArray && matchType == "type" =>
+          if (ra.arrayOrEmpty.length >= arrayMin) RuleMatchSuccess
           else RuleMatchFailure
 
-        case (Some(matchType), None, None) if matchType == "type" =>
+        case (Some(matchType), None, None) if ra.isArray && matchType == "type" =>
           RuleMatchSuccess
 
-        case (Some(matchType), None, None) if matchType == "regex" =>
+        case (Some(matchType), None, None) if ra.isArray && matchType == "regex" =>
           println("Regex match specified but no rule was supplied".yellow)
           RuleMatchFailure
 
-        case (Some(matchType), Some(regularExpression), None) if matchType == "regex" =>
-          if(ra.isString && ra.string.exists(_.matches(regularExpression))) RuleMatchSuccess
+        case (_, _, _) if ra.isObject =>
+          val e: Json = next.expected
+
+          val matching = PermissiveJsonEqualityHelper.areEqual(
+            Map(next.ruleAndContext.path -> next.ruleAndContext.rule),
+            e,
+            ra,
+            next.fieldName
+          )
+
+          if(matching) RuleMatchSuccess else RuleMatchFailure
```

Comment:
This section (above) is the real fix. Essentially the code does not cope with the scenario where you are doing wildcard based array matching and then you find an object. Permissive matching isn't really correct since you might have asked for strict but because of the way I've written this... it's too late. We've lost all context, and so we have this little hack.

Ultimately the matching logic is being rewritten and that's the real fix here.